main.js

```diff
@@ -1,18 +1,23 @@
 var $img = document.querySelector('img');
 
-document.onkeydown = function (event) {
-  switch (event.keyCode) {
-    case 37:
-      $img.className = 'west';
-      break;
-    case 38:
-      $img.className = 'north';
-      break;
-    case 39:
-      $img.className = 'east';
-      break;
-    case 40:
-      $img.className = 'south';
-      break;
+document.addEventListener('keydown', function rotate(event) {
+  if (event.keyCode === 40) {
+    $img.className = 'south';
+  } else if (event.keyCode === 37) {
+    $img.className = 'west';
+  } else if (event.keyCode === 38) {
+    $img.className = 'north';
+  } else if (event.keyCode === 39) {
+    $img.className = 'east';
+  } else if (event.keyCode === 32) {
+    setInterval(moveToRight, 16);
   }
-};
+});
+
+function moveToRight() {
+  var num = 0;
+  for (var i = 0; i < 100; i++) {
+    num = i;
+    var valueOfStyle = 'left:' + num + '%';
+  } return $img.setAttribute('style', valueOfStyle);
+}
```

Comment:
You should not be incrementing by a percentage, you should be incrementing by pixels, just a few every 16 milliseconds. Also, by using a `for` loop, this is happening all at once. You just end up setting `left: 100%` right away every time.